src/SDK/ConfigBuilder.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK;
+
+use Nette\Schema\Expect;
+use Nette\Schema\Processor;
+
+/** @phan-file-suppress PhanUndeclaredClassMethod */
+
+class ConfigBuilder
```

Comment:
I think this ConfigBuilder is doing to much. It's a) defining a configuration schema, and b) acting upon user definded values via env vars. I think the providers you outlined [here](https://github.com/open-telemetry/opentelemetry-php/issues/301#issuecomment-984091668) are a cleaner approach to deal with user provided values and allow for single responsibility.